domain/domain.go

```diff
@@ -254,121 +254,125 @@ func (do *Domain) GetScope(status string) variable.ScopeFlag {
 	return variable.DefaultScopeFlag
 }
 
-func (do *Domain) tryReload() {
-	// If we don't have update the schema for a long time > lease, we must force reloading it.
-	// Although we try to reload schema every lease time in a goroutine, sometimes it may not run accurately.
-	// e.g., the machine has a very high load, running the ticker is delayed.
-	last := atomic.LoadInt64(&do.lastLeaseTS)
-	lease := do.ddl.GetLease()
-
-	// if lease is 0, we use the local store, so no need to reload.
-	if lease > 0 && time.Now().UnixNano()-last > lease.Nanoseconds() {
-		do.MustReload()
+func (do *Domain) mockReloadFailed() error {
+	ver, err := do.store.CurrentVersion()
+	if err != nil {
+		log.Errorf("mock reload failed err:%v", err)
+		return errors.Trace(err)
 	}
+	lease := do.DDL().GetLease()
+	// Make sure that is timed out when checking validity.
+	mockLastSuccTime := time.Now().UnixNano() - int64(lease)
+	log.Warnf("mock lastSuccTS:%v, lease:%v", time.Now(), time.Duration(lease))
+	do.SchemaValidity.updateTimeInfo(mockLastSuccTime, ver.Ver)
+	return errors.New("mock reload failed")
 }
 
-var defaultMinReloadTimeout = 20 * time.Second
+const doReloadSleepTime = 500 * time.Millisecond
 
 // Reload reloads InfoSchema.
+// It's public in order to do the test.
 func (do *Domain) Reload() error {
```

Comment:
I think sleep and retry in Reload is not a good idea. 
There would be new ticker signal happen during this, and may cause Reload be called several times.

A better way is reload returns error and let `loadSchemaInLoop` handle the retry.